src/main/java/org/opensearch/plugin/insights/core/service/categorizer/QueryShapeGenerator.java

```diff
@@ -38,13 +38,22 @@ public class QueryShapeGenerator implements ClusterStateListener {
     static final String EMPTY_STRING = "";
     static final String ONE_SPACE_INDENT = " ";
     private final ClusterService clusterService;
-    private final String NO_FIELD_TYPE_VALUE = "";
     private final IndicesFieldTypeCache indicesFieldTypeCache;
+    private long cacheHitCount;
+    private long cacheMissCount;
+
+    private final String NO_FIELD_TYPE_VALUE = "";
+    public static final String HITS_STRING = "hits";
+    public static final String MISSES_STRING = "misses";
+    public static final String BYTES_STRING = "bytes";
```

Comment:
Should we have stat for number of entries as well?

Reply:
Added count